src/plugins/analysis/file_system_metadata/code/file_system_metadata.py

```diff
@@ -270,3 +315,10 @@ def _get_tar_file_mode_str(file_info: tarfile.TarInfo) -> str:
 
 def _has_correct_type(mime_type: str) -> bool:
     return mime_type in ARCHIVE_MIME_TYPES + FS_MIME_TYPES
+
+
+def _filemode_str_to_int(filemode: str) -> int:
```

Comment:
I believe this function is buggy. It returns an invalid octal value for, e.g., `-rwsr-xr-x`

```python
In [6]: oct(filemode_str_to_int("-rwsr-xr-x"))
Out[6]: '0o104755'
```

It should be `0o4755`.

Reply:
Unix permissions are limited to four octal digits.

These bad permissions are persisted in the plugin results.

Reply:
Okay. So for reference: the `filemode` is correct! Because the [leftmost two octals describe the file type](https://stackoverflow.com/a/4087441).
However, this value describes more than a `permission`. We decided to create a new PR that introduces a new analysis result field that separates `permissions` from `file types`.